src/server/routes/unitReadings.js

```diff
@@ -494,25 +486,80 @@ function createRouter() {
 		if (!(validateGroupThreeDReadingsParams(req.params) && validateThreeDQueryParams(req.query))) {
 			res.sendStatus(400);
 		} else {
-			// Get time range to validate 1 year or less.
+			// Get time range - auto-adjustment handled on frontend
 			const timeInterval = TimeInterval.fromString(req.query.timeInterval);
 			if (!timeInterval.getIsBounded()) {
 				// Cannot do if not bounded.
 				res.sendStatus(400);
 			} else {
-				const duration = moment.duration(timeInterval.endTimestamp.diff(timeInterval.startTimestamp));
-				// Gets 0 unless one day beyond a year but that okay since don't do partial days.
-				const durationInYears = duration.years();
-				if (durationInYears >= 1) {
-					// Limit 3D to one year of data.
-					res.sendStatus(400);
+				const groupID = req.params.group_id;
+				const graphicUnitID = req.query.graphicUnitId;
+				const readingInterval = req.query.readingInterval;
+				const forJson = await groupThreeDReadings(groupID, graphicUnitID, timeInterval, readingInterval);
```

Comment:
This is an open route (login not needed). Thus, a bad actor could request arbitrary readingInterval and return a lot of data. Before it limited it to 1 year but that is no longer done. The range needs to be limited to the desired value. Applies to meter too.